packages/graph/hash_graph/lib/graph/src/shared/identifier.rs

```diff
@@ -0,0 +1,57 @@
+use serde::{Deserialize, Serialize, Serializer};
+use serde_json;
+use type_system::uri::VersionedUri;
+use utoipa::{openapi, ToSchema};
+
+use crate::knowledge::EntityId;
+
+// TODO - This is temporary and introduced for consistency, we need to introduce actual IDs for
+//  links, should be revisited as part of https://app.asana.com/0/0/1203157172269853/f
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct LinkId {
+    pub source_entity_id: EntityId,
+    pub target_entity_id: EntityId,
+    #[schema(value_type = String)]
+    pub link_type_id: VersionedUri,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum GraphElementIdentifier {
+    OntologyElementId(VersionedUri),
+    // TODO: owned_by_id and version are required to identify a specific instance of an entity
+    //  https://app.asana.com/0/1202805690238892/1203214689883091/f
+    KnowledgeGraphElementId(EntityId),
+    Temporary(LinkId),
+}
+
+impl Serialize for GraphElementIdentifier {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self {
+            Self::KnowledgeGraphElementId(identifier) => identifier.serialize(serializer),
+            Self::OntologyElementId(identifier) => identifier.serialize(serializer),
+            Self::Temporary(identifier) => serializer.serialize_str(&format!(
+                "<SOURCE>{}<TARGET>{}<TYPE>{}",
```

Comment:
Any reason for this bracketed delimiter style output? 
Is it on purpose that the serializers for `KnowledgeGraphElementId` and `OntologyElementId` both don't have the same style of delimiters? 
(I appreciate this is temporary and that we want to introduce `(entity_id, owned_by_id)` instead for entity identifiers, just wondering if it could be problematic when supplying malicious hand-crafted IDs right now)

Reply:
Purely because it's temporary and I needed to put something in there. Underscores or other single-character delimiters seemed too easy to break so I just went for something a bit verbose.

> Is it on purpose that the serializers for KnowledgeGraphElementId and OntologyElementId both don't have the same style of delimiters?

They're single elements and have nothing to delimit 😁 

I definitely think this is something we should figure out properly, but I'd propose we ignore it for now, because there are follow-ups planned to completely rework how we define identifiers for elements, and the only one here that's different is something that's actively not used.